File: leetcode2/1easy/유시연/999.py
class Solution:
    def numRookCaptures(self, board: List[List[str]]) -> int:
        
        count = 0
        rook = (0, 0)
        for i in range(8):
            for j in range(8):
                if board[i][j] == 'R':
                    rook = (i,j)
                    break
                    
        R_Y = rook[1]
        R_X = rook[0]
        while R_X >= 0:
            if board[R_X][R_Y] == 'B':
                break
            if board[R_X][R_Y] == 'p':
                count += 1
                break
            R_X -= 1

        R_X = rook[0]
        while R_X < 8:
            if board[R_X][R_Y] == 'B':
                break
            if board[R_X][R_Y] == 'p':
                count += 1
                break
            R_X += 1
        
        R_X = rook[0]
        R_Y = rook[1]
        while R_Y >= 0:
            if board[R_X][R_Y] == 'B':
                break
            if board[R_X][R_Y] == 'p':
                count += 1
                break
            R_Y -= 1

        R_Y = rook[1]
        while R_Y < 8:
            if board[R_X][R_Y] == 'B':
                break
            if board[R_X][R_Y] == 'p':
                count += 1
                break
            R_Y += 1
        
        return count
                    
                
            
        

File: leetcode2/1easy/이준열/999.cpp
class Solution {
    public:
        int dirs[4][2] = {{1, 0}, {-1,0}, {0,1}, {0,-1}};
        
    
        int numRookCaptures(vector<vector<char>>& board) {
            pair<int, int> RookLoc;
    
            for (int i = 0 ; i < 8; i++)
            {
                for (int j = 0; j < 8; j++)
                {
                    if (board[i][j] == 'R')
                        RookLoc = make_pair(i, j);
                }
            }
    
            int output = 0;
    
            for (int d = 0; d < 4; d++)
            {
                int m = RookLoc.first;
                int n = RookLoc.second;
    
                while (((m >= 0) && (m < 8)) &&
                        ((n >= 0) && (n < 8)))
                {
                    if (board[m][n] == 'B')
                        break;
                    if (board[m][n] == 'p')
                    {
                        output++;
                        break;
                    }
                    m += dirs[d][0];
                    n += dirs[d][1];
                }
            }
    
            return output;
            
        }
    };